include/zephyr/spinlock.h

```diff
@@ -183,7 +192,16 @@ static ALWAYS_INLINE void k_spin_unlock(struct k_spinlock *l,
 	ARG_UNUSED(l);
 #ifdef CONFIG_SPIN_VALIDATE
 	__ASSERT(z_spin_unlock_valid(l), "Not my spinlock %p", l);
-#endif
+
+#if defined(CONFIG_SPIN_LOCK_TIME_LIMIT) && (CONFIG_SPIN_LOCK_TIME_LIMIT != 0)
+	/* Verify the time limit has not been exceeded ignoring
+	 * the rule if there was a counter wrap.
+	 */
+	__ASSERT(k_us_to_cyc_floor32(CONFIG_SPIN_LOCK_TIME_LIMIT) >
```

Comment:
That conversion is runtime code on some platforms (really just HPET I think, where the cycle rate is probed and unknown at build time), which turns this into a pretty heavyweight bit of intstrumentation.  Might be cleaner just to specify it in cycles.  Real time units are nice when you can have them, but anyone who cares about tuning at this level will know how to read a cycle time.